File: app/src/main/java/com/mortarnav/Parceler.java
package com.mortarnav;

import android.os.Parcelable;

import org.parceler.Parcels;

import architect.ScreenPath;
import architect.ScreenParceler;

/**
 * PathParceler that uses Parceler library
 *
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
public class Parceler implements ScreenParceler {

    @Override
    public Parcelable wrap(ScreenPath path) {
        return Parcels.wrap(path);
    }

    @Override
    public ScreenPath unwrap(Parcelable parcelable) {
        return Parcels.unwrap(parcelable);
    }
}

File: kotlin_app/src/main/kotlin/architect/kotlinapp/Parceler.kt
package architect.kotlinapp

import android.os.Parcelable
import architect.ScreenParceler
import architect.ScreenPath
import nz.bradcampbell.paperparcel.PaperParcels
import nz.bradcampbell.paperparcel.TypedParcelable

/**
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
class Parceler : ScreenParceler<ScreenPath> {

    override fun wrap(path: ScreenPath?): Parcelable? {
        return PaperParcels.wrap(path)
    }

    override fun unwrap(parcelable: Parcelable?): ScreenPath? {
        return PaperParcels.unwrap(parcelable as TypedParcelable<ScreenPath>)
    }
}